includes/class-amp-theme-support.php

```diff
@@ -275,6 +279,44 @@ public static function register_content_embed_handlers() {
 		return $embed_handlers;
 	}
 
+	/**
+	 * Add the comments template placeholder marker
+	 *
+	 * @param array $args the args for the comments list..
+	 * @return array Args to return.
+	 */
+	public static function amp_set_comments_walker( $args ) {
+		$amp_walker     = new AMP_Comment_Walker();
+		$args['walker'] = $amp_walker;
+		// @todo Make threaded work by setting 'data-update-time' of the parent first level parent with the last timestamp.
+		$args['max_depth'] = 1;
+
+		return $args;
+	}
+
+	/**
+	 * Adds the form submit success and fail templates.
+	 *
+	 * @param string $post_id The post ID if action is 'comment_form'.
+	 */
+	public static function add_amp_comment_form_templates( $post_id ) {
+		$output = '';
+		if ( empty( $post_id ) ) {
+			$output .= '<div id="amp-comment-form-fields">';
+		} else {
+			$output .= '</div>';
+			$output .= '<div submit-success><template type="amp-mustache">';
+			$output .= esc_html__( 'Your comment has been posted.', 'amp' );
+			$output .= sprintf( '<a class="amp-view-new-comment" href="{{comment_link}}">%s</a>', esc_html__( 'View Comment', 'amp' ) );
+			$output .= '</div></template>';
+			$output .= '<div submit-error><template type="amp-mustache">';
+			$output .= '<p class="amp-comment-submit-error">{{{error}}}</p>';
+			$output .= '</div></template>';
```

Comment:
Instead of creating a string of HTML and then echoing it, how about just `?><div>...</div><?php`?

Reply:
The idea was to have it pass through a filter before outputting it. Would that be a good idea?

Reply:
I don't think we need to add a filter here? In any case, output buffering could be used here to capture the string. On the other hand, the HTML could also get loaded from an external template file which could potentially be overridden by a theme. I'd have to refresh my memory on how comment forms are normally extended.